addons/mail/static/src/models/notification_list_view/notification_list_view.js

```diff
@@ -0,0 +1,222 @@
+        _computeThreadNeedactionPreviewViews() {
+            if (this.filter !== 'all') {
+                return clear();
+            }
+            return insertAndReplace(
+                this.messaging.models['mail.thread']
+                    .all(t => t.model !== 'mail.box' && t.needactionMessagesAsOriginThread.length > 0)
+                    .sort((t1, t2) => {
+                        if (t1.needactionMessagesAsOriginThread.length > 0 && t2.needactionMessagesAsOriginThread.length === 0) {
+                            return -1;
+                        }
+                        if (t1.needactionMessagesAsOriginThread.length === 0 && t2.needactionMessagesAsOriginThread.length > 0) {
+                            return 1;
+                        }
+                        if (t1.lastNeedactionMessageAsOriginThread && t2.lastNeedactionMessageAsOriginThread) {
+                            return t1.lastNeedactionMessageAsOriginThread.id < t2.lastNeedactionMessageAsOriginThread.id ? 1 : -1;
+                        }
+                        if (t1.lastNeedactionMessageAsOriginThread) {
+                            return -1;
+                        }
+                        if (t2.lastNeedactionMessageAsOriginThread) {
+                            return 1;
+                        }
+                        return t1.id < t2.id ? -1 : 1;
+                    })
+                    .map(thread => {
+                        return { thread: replace(thread) };
+                    })
+            );
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeThreadPreviewViews() {
+            const threads = this._getThreads();
+            // thread notifications
+            return insertAndReplace(
+                threads
+                    .sort((t1, t2) => {
+                        if (t1.localMessageUnreadCounter > 0 && t2.localMessageUnreadCounter === 0) {
+                            return -1;
+                        }
+                        if (t1.localMessageUnreadCounter === 0 && t2.localMessageUnreadCounter > 0) {
+                            return 1;
+                        }
+                        if (t1.lastMessage && t2.lastMessage) {
+                            return t1.lastMessage.id < t2.lastMessage.id ? 1 : -1;
+                        }
+                        if (t1.lastMessage) {
+                            return -1;
+                        }
+                        if (t2.lastMessage) {
+                            return 1;
+                        }
+                        return t1.id < t2.id ? -1 : 1;
+                    })
+                    .map(thread => {
+                        return {
+                            thread: replace(thread),
+                        };
+                    })
+            );
+        },
+        /**
+         * @private
+         * @returns {mail.thread[]}
+         */
+        _getThreads() {
```

Comment:
Could turn this into a field

Reply:
It's not super important, but considering the amount of changes, could be worth making things easier to maintain like turning this into field.

Reply:
Yes I've done the change, it's also better for performance to avoid recomputing all threads every time something else changes.